File: backend/blackbox/src/main/java/com/insight_web/blackbox/usecase/SiteCreation.java
package com.insight_web.blackbox.usecase;

import com.insight_web.blackbox.domain.Site;
import com.insight_web.blackbox.domain.User;
import com.insight_web.blackbox.usecase.action.SiteAction;
import com.insight_web.blackbox.usecase.storage.SiteStorage;
import com.insight_web.blackbox.usecase.storage.UserStorage;
import jakarta.transaction.Transactional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class SiteCreation {
    private final UserStorage userStorage;
    private final SiteStorage siteStorage;

    @Transactional
    public Site create(SiteAction siteAction, String userEmail) {
        var userId = userStorage.findByEmail(userEmail).getId();
        var site = siteAction.toDomainModel(userId);

        return siteStorage.create(site);
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/usecase/SiteCreation.kt
package com.insightweb.auth.usecase

import com.insightweb.auth.domain.Site
import com.insightweb.auth.usecase.action.SiteAction
import com.insightweb.auth.usecase.storage.SiteStorage
import com.insightweb.auth.usecase.storage.UserStorage
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
class SiteCreation(
    private val userStorage: UserStorage,
    private val siteStorage: SiteStorage,
) {
    @Transactional
    fun create(
        siteAction: SiteAction,
        userEmail: String
    ): Site {
        val userId: Long = userStorage.findByEmail(userEmail).id!!
        val site: Site = siteAction.toSite(userId)
        return siteStorage.create(site)
    }
}